stdlib/Random/src/misc.jl

```diff
@@ -71,13 +71,14 @@ let b = UInt8['0':'9';'A':'Z';'a':'z']
     global randstring
 
     function randstring(r::AbstractRNG, chars=b, n::Integer=8)
+        _n = convert(Int, n)
```

Comment:
I am not sure it is worth doing in this specific instance but presumably structuring it like:

```julia
f(x::Integer) = f(convert(Int, n))
f(x::Int) = ...
```

would reduce compilation time since you would not compile the main function body for multiple integer types.

Reply:
For cases where lots of different types are passed to `randstring`, that's certainly the case, yeah. I'm not sure how common that would be though, there's lots of better ways to control the length of a generated string other than through the type.

Reply:
I think @KristofferC is just referring to the amount of code that is generated by using function barriers or: 
https://docs.julialang.org/en/v1/manual/performance-tips/#kernel-functions

Reply:
Yes, I'm referring to the likelihood of hitting that & the additional compilation overhead being a significant slowdown. If this is a bottleneck, I'd first recommend switching to a different scheme for generating the length than optimize the compilation overhead.

Reply:
The irony appears lost on you, that this PR would allow someone to have a type unstable call site with various `Integer` types for length, rather than error as it currently does.

Reply:
Why would that inherently be a problem? We don't prevent type instabilities in user code in other places either.

Reply:
I think Krostoffer's convert approach is a good idea. Yes, it's ok to just allow a different integer type flow through but in this case Int covers the entire range of reasonable values and converting avoids additional compilation.

Reply:
There is no other integer type flowing through though; the code in this PR already calls `_convert(Int, ..)`. This function is a total of 11 lines long, I seriously doubt that this leads to a bottleneck in either compilation time or binary size..

Reply:
Ok, but every little bit counts. Why not just change it?

Reply:
Because I agree with the comment by Kristoffer that it's probably not worth doing in this case. Is it a problem that I agree with the reviewer? As it stands, this just seems like premature optimization of a hypothetical to me.